main.tf

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/******************************************
+	Scheduled Function Definition
+ *****************************************/
+
+resource "google_cloud_scheduler_job" "job" {
+  provider = "google-beta"
+  name     = "${var.job_name}"
+  description = "${var.job_description}"
+  schedule = "${var.job_schedule}"
+
+  pubsub_target = {
+    topic_name = "projects/${var.project_id}/topics/${module.pubsub_topic.topic}"
+    data = "${var.message_data}"
+  }
+}
+
+/******************************************
+	PubSub Topic Definition
+ *****************************************/
+
+module "pubsub_topic" {
+  source  = "github.com/terraform-google-modules/terraform-google-pubsub"
```

Comment:
If you're calling out to another module within a module, definitely pin it to a version with `?ref=vX.Y.Z`.

Reply:
I didn't realize that was possible, good call.